src/client/coll.rs

```diff
@@ -0,0 +1,201 @@
+use bson;
+use client::db::Database;
+use client::common::{ReadPreference, WriteConcern};
+use client::wire_protocol::operations::{OpQueryFlags, Message};
+
+/// Interfaces with a MongoDB collection.
+pub struct Collection<'a> {
+    db: &'a Database<'a>,
+    pub name: String,
+    pub full_name: String,
+    read_preference: Option<ReadPreference>,
+    write_concern: Option<WriteConcern>,
+}
+
+#[derive(Clone, PartialEq)]
+pub enum CursorType {
+    NonTailable,
+    Tailable,
+    TailableAwait,
+}
+
+#[derive(Clone)]
+pub struct AggregateOptions {
+    pub allow_disk_use: bool,
+    pub use_cursor: bool,
+    pub batch_size: Option<i32>,
+    pub max_time_ms: Option<i64>,
+}
+
+#[derive(Clone)]
+pub struct CountOptions {
+    pub hint: Option<bson::Document>,
+    pub limit: Option<i64>,
+    pub max_time_ms: Option<i64>,
+    pub skip: Option<i64>,
+}
+
+#[derive(Clone)]
+pub struct DistinctOptions {
+    pub max_time_ms: Option<i64>,
+}
+
+#[derive(Clone)]
+pub struct FindOptions {
+    pub allow_partial_results: bool,
+    pub no_cursor_timeout: bool,
+    pub op_log_replay: bool,
+    pub skip: i32,
+    pub limit: i32,
+    pub cursor_type: CursorType,
+    pub batch_size: Option<i32>,
+    pub comment: Option<String>,
+    pub max_time_ms: Option<i64>,
+    pub modifiers: Option<bson::Document>,
+    pub projection: Option<bson::Document>,
+    pub sort: Option<bson::Document>,
+}
+
+impl FindOptions {
+    /// Creates a new FindOptions struct with default parameters.
+    pub fn new() -> FindOptions {
+        FindOptions {
+            allow_partial_results: false,
+            no_cursor_timeout: false,
+            op_log_replay: false,
+            skip: 0,
+            limit: 0,
+            cursor_type: CursorType::NonTailable,
+            batch_size: None,
+            comment: None,
+            max_time_ms: None,
+            modifiers: None,
+            projection: None,
+            sort: None,
+        }
+    }
+
+    pub fn with_skip(&self, skip: i32) -> FindOptions {
```

Comment:
This method (as well as `with_projection`) could probably be changed to take `&mut self` as an argument instead of returning a cloned struct.